### ISSUE TITLE
Allow use of "serde-lite" with server functions

### ISSUE THREAD

**Issue by janschiefer** [leptos-rs/leptos#1501]
**Is your feature request related to a problem? Please describe.**
It seems "serde-lite" is currently only supported incompletely.

If I use the "serde-lite" feature with server functions, the "server" macro won't compile:

```
error[E0463]: can't find crate for `serde`
  --> src/api/item_details_data_api.rs:31:1
   |
31 | #[server(GetItemDetailsData, "/api", "GetJson", "get_item_details_data")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate
   |
   = note: this error originates in the derive macro `::leptos::server_fn::serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

```

**Describe the solution you'd like**t
Reweite the "server" macro to use the serde-lite crate, when the feature "serde-lite" is enabled. 

**Describe alternatives you've considered**
Minisede seems to have similiar problems.

**Additional context**
...

**Comment by gbj** [comment]
`serde-lite` and `miniserde` are used for serializing `create_resource` between the server and the client.

They could hypothetically also be used for serializing the *results* of server functions, but `serde` would still be required for serializing the *arguments*, because `serde-lite` and `miniserde` only support serializing to JSON, while the arguments are serialized to URL-encoded data using `serde_qs`, which requires `serde`.

For the same reasons of progressive enhancement, `<form>` support, and complexity described in #1409 and #1159 with regard to additional HTTP verbs in server functions, I don't really intend to add support for additional, non-URL-encoded arguments.

I'd happily take a PR for `serde-lite` or `miniserde` support for server function return types, but it's probably not something I'll have time to work on myself.

**Comment by rakshith-ravi** [comment]
Potentially fixed by #2168

**Comment by gbj** [comment]
Added in #2158.
